src/read/read_ref.rs

```diff
@@ -40,6 +40,14 @@ pub trait ReadRef<'a>: Clone + Copy {
     /// Returns an error if offset or size are out of bounds.
     fn read_bytes_at(self, offset: u64, size: u64) -> Result<&'a [u8]>;
 
+    /// Get a reference to a delimited `u8` slice at the given offset.
+    ///
+    /// Does not include the delimiter.
+    ///
+    /// Returns an error if offset is out of bounds or the delimiter is
+    /// not found.
+    fn read_bytes_at_until(self, offset: u64, terminator: u8) -> Result<&'a [u8]>;
```

Comment:
Comment and argument name are inconsistent about whether it's called delimiter or terminator